### DISCUSSION TITLE
Git actions suddenly failing when running npm ci? npm ci works locally

### DISCUSSION THREAD

**Discussion by iddqdRepo** [community/community#33852] [Actions] [answered]
This is a new behaviour, it has worked up until a few days ago.

When I run npm install, remove or add any package (or do anything that changes the local package.json) and commit to a PR branch, I get the following error in the github actions "run npm ci" build.

```
Run npm ci
  
npm ERR! code EUSAGE
npm ERR! 
npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm ERR! 
npm ERR! Missing: webpack@5.7[4](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:4).0 from lock file
npm ERR! Missing: @types/eslint-scope@3.7.[4](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:5) from lock file
npm ERR! Missing: @types/estree@0.0.[5](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:6)1 from lock file
npm ERR! Missing: @xtuc/long@4.2.2 from lock file
npm ERR! Missing: @webassemblyjs/helper-buffer@1.11.1 from lock file
npm ERR! Missing: @webassemblyjs/helper-wasm-section@1.11.1 from lock file
npm ERR! Missing: @webassemblyjs/wasm-gen@1.11.1 from lock file
npm ERR! Missing: @webassemblyjs/wasm-opt@1.11.1 from lock file
npm ERR! Missing: @webassemblyjs/wast-printer@1.11.1 from lock file
npm ERR! Missing: @webassemblyjs/ieee[7](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:8)54@1.11.1 from lock file
npm ERR! Missing: commander@2.[20](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:21).3 from lock file
npm ERR! Missing: source-map-support@0.5.[21](https://github.com/iddqdRepo/brightEyesNextJsUpgrade/actions/runs/3106889444/jobs/5034329266#step:5:22) from lock file
npm ERR! Missing: @jridgewell/gen-mapping@0.3.2 from lock file
npm ERR! Missing: buffer-from@1.1.2 from lock file
npm ERR! Missing: source-map@0.6.1 from lock file
npm ERR! Missing: estraverse@4.3.0 from lock file
npm ERR! 
npm ERR! Clean install a project
```

**There are a lot more missing packages, I removed some as to not spam**

My main.yml file:
```
name: Node.js CI

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

jobs:
  build:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [16.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:
      - uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
          cache: "npm"
      - run: npm ci
      - run: npm run lint
      - run: npm run check-format
      - run: npm run build
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          start: npm start
```

Things i've tried:

- Deleting node modules & package-lock.json & running npm i - DOESNT WORK
- Npm cache-clean/audit fix - DOESNT WORK
- Upgrading my npm version  - DOESNT WORK
- Changing my github main.yml to npm ci --legacy-peer-deps - WORKS but causes other weird bugs about paths to other React components withing my project to not be found with a Type error: Cannot find module '../ etc" (they all work fine if the old package-lock is used)
- Pasting the old package-lock.json to overwrite my local copy and committing - WORKS But I can never actually remove or add packages.
- Adding run: npm Install before the run: npm ci  in main.yml - WORKS, but does this defeat the purpose?

In summary: If I create a new branch from main > npm install a new package & push the updated package.json and package-lock.json to the branch > it fails the github actions check on npm ci with the above npm ERR because the package-lock.json does not match.

I even tried going back to a bare bones project with just the pipeline set up, making a branch and i'm instantly getting the same issue 

I'm stumped, any help would be greatly appreciated.

**Comment by iddqdRepo** [comment, marked as answer]
Fixed:
Something went wrong with my local installation of node & npm. 
Had to remove all traces of npm & node from my pc and install with nvm
Followed these instructions for removal:
https://stackoverflow.com/questions/20711240/how-to-completely-remove-node-js-from-windows
